src/webui/api/apicontroller.h

```diff
@@ -67,6 +74,9 @@ class APIController : public QObject
     void setResult(const QJsonObject &result);
 
 private:
+    virtual bool isActionSafe(const QString &actionName) const;
```

Comment:
I think what you really want is declaring it a pure virtual function:
`virtual bool isActionSafe(const QString &actionName) const = 0;`
And remove the function definitions in .cpp file.